tools/exporters/data_dictionary_builder.php

```diff
@@ -36,100 +36,102 @@
  * Usage:
  *    php data_dictionary_builder.php
  *
- * PHP Version 5
+ * PHP Version 5,7
  *
  * @category Behavioural
  * @package  Main
  * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
  * @license  Loris License
  * @link     https://github.com/aces/Loris
  */
+require_once __DIR__ . '/../generic_includes.php';
 
-set_include_path(get_include_path().":../project/libraries:../php/libraries:");
+$parameter_types = array();
+$instrumentParameterTypeCategoryIDs = array();
+$instrumentParameterTypeIDs         = array();
 
-require_once __DIR__ . "/../vendor/autoload.php";
-$configFile = "../project/config.xml";
-$client     = new NDB_Client();
-$client->makeCommandLine();
-$client->initialize($configFile);
-$DB = Database::singleton();
 
-// Set a user for the history table
-if (!isset($_SESSION['State'])) {
-    $_SESSION['State'] =& State::singleton();
-}
-$_SESSION['State']->setUsername(get_current_user());
-
-// define which configuration file we're using for this installation
-//Get the entries we already have in the DB
-getColumns(
+$parameter_types = $DB->pselectColWithIndexKey(
     "Select Name, ParameterTypeID from parameter_type",
-    $DB,
-    $parameter_types,
-    array()
+    array(),
+    "Name"
 );
 
-//Delete in the parameter_type table relating to
-//parameter_type_category.Type = 'Instrument', without affecting other entries.
-//get parameter_type_category.Type ='Instrument' ParamenterTypeCategoryIDs
-getColumn(
-    "SELECT ParameterTypeCategoryID
-        FROM parameter_type_category
-        WHERE Type = :in",
-    $DB,
-    $instrumentParameterTypeCategoryIDs,
-    array('in' => 'Instrument')
+
+//Get all category IDs for instruments
+$instrumentParameterTypeCategoryIDs = $DB->pselectCol(
+    "SELECT ParameterTypeCategoryID 
+            FROM parameter_type_category 
+            WHERE Type = :in",
+    array("in" => "Instrument")
 );
 
+// DELETING all parameter_type data associated to instruments
 if (!empty($instrumentParameterTypeCategoryIDs)) {
-    // Prepare for the IN () SQL quandary
+    $instrumentParameterTypeIDString         = "";
+    $instrumentParameterTypeCategoryIDString = "";
+
+    //Prepare for the 'IN ()' SQL statement to get parameter_type values
+    //associated with the instrument categories
     $prepIn = prepareIn($instrumentParameterTypeCategoryIDs);
 
-    //get all 'Instrument' ParameterTypeIDs
-    getColumn(
+    //Get all types IDs related to instrument categories
+    $instrumentParameterTypeIDs = $DB->pselectCol(
         "SELECT ParameterTypeID
             FROM parameter_type_category_rel
             WHERE ParameterTypeCategoryID
-            IN (" . $prepIn['query'] . ")",
-        $DB,
-        $instrumentParameterTypeIDs,
-        $prepIn['array']
+            IN (" . $prepIn["query"] . ")",
```

Comment:
@driusan should we be using find in set instead ??

Reply:
We should use DELETE JOIN